src/Microsoft.AspNetCore.Server.Kestrel/Http/Frame.cs

```diff
@@ -176,16 +176,28 @@ public bool HasResponseStarted
 
         public void InitializeHeaders()
         {
-            _frameHeaders = HttpComponentFactory.CreateHeaders(DateHeaderValueManager);
-            RequestHeaders = _frameHeaders.RequestHeaders;
-            ResponseHeaders = _frameHeaders.ResponseHeaders;
+            if(_frameHeaders == null)
+            {
+                _frameHeaders = new Headers();
+                _frameHeaders.Initialize(DateHeaderValueManager);
+                RequestHeaders = _frameHeaders.RequestHeaders;
+                ResponseHeaders = _frameHeaders.ResponseHeaders;
+            }
+            else
+            {
+                _frameHeaders.Initialize(DateHeaderValueManager);
```

Comment:
Can the call to `Initialize()` be moved after the `if` block, instead of being duplicated in both blocks?